plugins/crypto/mbedtls/ua_mbedtls_create_certificate.c

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2021 (c) Christian von Arnim, ISW University of Stuttgart (for VDW and umati)
+ *
+ */
+
+#include "securitypolicy_mbedtls_common.h"
+
+#ifdef UA_ENABLE_ENCRYPTION_MBEDTLS
+
+UA_StatusCode
+UA_CreateCertificate(const UA_Logger *logger,
+                    UA_String subject[], UA_UInt32 lenSubject,
+                    UA_String subjectAltName[], UA_UInt32 lenSubjectAltName,
+                    UA_ByteString *outPKey, UA_ByteString *outCert,
+                    enum UA_CertificateFormat certFormat) {
+    if(!outPKey || !outCert || !logger || !subjectAltName
+        || !subject || lenSubjectAltName == 0 || lenSubject == 0) {
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
```

Comment:
If it builds, then it should work.
I prefer to have this function undefined for mbedtls (linker error for a missing function) than to let it fail only at runtime.
That reduces the potential surprise at runtime.